persistence-api/src/main/java/io/stargate/db/schema/SchemaHash.java

```diff
@@ -0,0 +1,69 @@
+package io.stargate.db.schema;
+
+import java.util.Collection;
+import java.util.Objects;
+
+/** An interface for calculating deterministic hash codes for schema objects. */
+public interface SchemaHash {
+
+  /**
+   * Calculates a deterministic hash code for the schema object that does not change between
+   * invocations of the JVM. This was mostly because it was determined that hash codes of
+   * enums could change between different executions of the JVM and schema hash codes need to
+   * be deterministic across JVM instances because these codes are distributed to Stargate
+   * bridge clients.
+   *
+   * <p>When calculating the schema hash code it advised that one of the `SchemaHash#hashCode()`
+   * methods in this interface be used instead of using `obj.hashCode()`, {@link Object#hashCode()},
+   * or {@link System#identityHashCode(Object)}, etc. to avoid an error that produces a
+   * non-deterministic hash code.
+   *
+   * @return a deterministic hash code for the schema object.
+   */
+  int schemaHashCode();
+
+  static int hashCode(SchemaHash object) {
+    if (object == null) {
+      return 0;
+    }
+    return object.schemaHashCode();
+  }
+
+  static int hashCode(Object object) {
+    // It's still possible that something like `Optional<Enum>` would make it through. So that's
+    // something to look out for.
+    if (object instanceof Enum || object instanceof SchemaHash) {
```

Comment:
`object instanceof SchemaHash` seems not needed, you have up method `static int hashCode(SchemaHash object) ` so you would never the that here?

Reply:
It's slight protection from stuff like `SchemaHash.hashCode((Object)table)`.

Reply:
Okey